src/test/java/com/michelin/kafkactl/service/ResourceServicePrepareResourcesTest.java

```diff
@@ -0,0 +1,316 @@
+        Map<String, Resource> resourceMap = new HashMap<>();
+        for (Map.Entry<String, SchemaData> entry : filesMap.entrySet()) {
+            File file = entry.getValue().schemaTempFile;
+            String schemaContent = entry.getValue().schemaContent;
+            String schemaName = entry.getKey();
+            Files.writeString(file.toPath(), schemaContent);
+            Map<String, Object> spec = new HashMap<>(Map.of(SCHEMA_FILE, file.getAbsolutePath()));
+            var referencesMap = new ArrayList<Map<String, String>>();
+            for (String ref : entry.getValue().references) {
+                referencesMap.add(Map.of("name", ref));
+            }
+            spec.put("references", referencesMap);
+            Resource res = Resource.builder()
+                    .kind("Schema")
+                    .metadata(Metadata.builder().name(schemaName).build())
+                    .spec(spec)
+                    .build();
+            resourceMap.put(schemaName, res);
+        }
+        return resourceMap;
+    }
+
+    @Test
+    void prepareResources_shouldEnrichSchemaFromFile() throws Exception {
+
+        File tempSchema = File.createTempFile("test-schema", ".avsc");
+        String schemaContent = "{\"type\":\"record\",\"name\":\"Test\"}";
+        Files.writeString(tempSchema.toPath(), schemaContent);
+
+        Map<String, Object> spec = new HashMap<>();
+        spec.put(SCHEMA_FILE, tempSchema.getAbsolutePath());
+
+        Resource schemaResource = Resource.builder()
+                .kind("Schema")
+                .metadata(Metadata.builder().name("test").build())
+                .spec(spec)
+                .build();
+
+        List<Resource> resources = new ArrayList<>(List.of(schemaResource));
+
+        resourceService.prepareResources(resources, commandSpec);
+
+        assertEquals(schemaContent, schemaResource.getSpec().get("schema"));
+
+        tempSchema.delete();
+    }
+
+    @Test
+    void prepareResources_shouldHandleFileNotFound() {
+        String fakePath = "not-exist.avsc";
+        Map<String, Object> spec = new HashMap<>();
+        spec.put(SCHEMA_FILE, fakePath);
+
+        Resource schemaResource = Resource.builder()
+                .kind("Schema")
+                .metadata(Metadata.builder().name("test").build())
+                .spec(spec)
+                .build();
+
+        Exception ex = assertThrows(
+                CommandLine.ParameterException.class,
+                () -> resourceService.prepareResources(List.of(schemaResource), commandSpec));
+
+        assertEquals(
+                "Cannot open schema file not-exist.avsc. Schema path must be relative to the CLI.", ex.getMessage());
+        assertNull(schemaResource.getSpec().get("schema"));
+    }
+
+    @Test
+    void prepareSchemaResources_shouldEnrichSchemasWithUnionDependency() throws Exception {
+        String schema1 = "{\"type\":\"record\",\"name\":\"Test1\",\"namespace\":\"com.example.one\"}";
```

Comment:
To make the test more readable, I think we can create the files physically in `src/test/resources/schemas` (we already have a `topic.yml`). It will avoid the `createTempFile` and `delete()` in the test